internal/controllers/gateway/gateway_controller.go

```diff
@@ -271,11 +271,13 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// to be gone at this point in which case it will be ignored.
 	gateway := new(gatewayv1beta1.Gateway)
 	if err := r.Get(ctx, req.NamespacedName, gateway); err != nil {
+		gateway.Namespace = req.Namespace
+		gateway.Name = req.Name
```

Comment:
I believe this should be placed inside the if branch below right? It's only used there

Reply:
done. I thought it was used in both branches, but when I changed the not `NotFound` branch, it is not used anymore in this branch.